kubernetes/resource_kubernetes_pod_disruption_budget.go

```diff
@@ -0,0 +1,202 @@
+package kubernetes
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	api "k8s.io/api/policy/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	pkgApi "k8s.io/apimachinery/pkg/types"
+	kubernetes "k8s.io/client-go/kubernetes"
+)
+
+// Use generated swagger docs from kubernetes' client-go to avoid copy/pasting them here
+var (
+	podDisruptionBudgetSpecDoc               = api.PodDisruptionBudget{}.SwaggerDoc()["spec"]
+	podDisruptionBudgetSpecMaxUnavailableDoc = api.PodDisruptionBudget{}.SwaggerDoc()["maxUnavailable"]
+	podDisruptionBudgetSpecMinAvailableDoc   = api.PodDisruptionBudget{}.SwaggerDoc()["minAvailable"]
+	podDisruptionBudgetSpecSelectorDoc       = api.PodDisruptionBudget{}.SwaggerDoc()["selector"]
+)
+
+func resourceKubernetesPodDisruptionBudget() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceKubernetesPodDisruptionBudgetCreate,
+		Read:   resourceKubernetesPodDisruptionBudgetRead,
+		Update: resourceKubernetesPodDisruptionBudgetUpdate,
+		Delete: resourceKubernetesPodDisruptionBudgetDelete,
+		Exists: resourceKubernetesPodDisruptionBudgetExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"metadata": namespacedMetadataSchema("pod disruption budget", true),
+			// Updates to spec not allowed; have to delete and recreate
+			// https://github.com/kubernetes/kubernetes/issues/45398
+			"spec": {
+				Type:        schema.TypeList,
+				Description: podDisruptionBudgetSpecDoc,
+				Required:    true,
+				ForceNew:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"max_unavailable": {
+							Type:         schema.TypeString,
+							Description:  podDisruptionBudgetSpecMaxUnavailableDoc,
+							Optional:     true,
+							ForceNew:     true,
+							ValidateFunc: validateTypeStringNullableIntOrPercent,
+						},
+						"min_available": {
+							Type:         schema.TypeString,
+							Description:  podDisruptionBudgetSpecMinAvailableDoc,
+							Optional:     true,
+							ForceNew:     true,
+							ValidateFunc: validateTypeStringNullableIntOrPercent,
+						},
+						"selector": {
+							Type:        schema.TypeList,
+							Description: podDisruptionBudgetSpecSelectorDoc,
+							Optional:    true,
+							ForceNew:    true,
+							MaxItems:    1,
+							Elem: &schema.Resource{
+								Schema: labelSelectorFields(false),
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceKubernetesPodDisruptionBudgetUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*kubernetes.Clientset)
+
+	namespace, name, err := idParts(d.Id())
+	if err != nil {
+		return err
+	}
+
+	ops := patchMetadata("metadata.0.", "/metadata/", d)
+	if d.HasChange("spec") {
+		specOps, err := patchPodDisruptionBudgetSpec("spec.0.", "/spec", d)
```

Comment:
Because the `spec` attribute is marked as `ForceNew` changes to it will always trigger a Delete/Create cycle rather than an Update.
This means that this patch operation is never going to get invoked and thus not needed.

Reply:
Done